src/frequenz/client/dispatch/types.py

```diff
@@ -148,7 +148,7 @@ def to_protobuf(self) -> PBRecurrenceRule.EndCriteria:
 
 
 # pylint: disable=too-many-instance-attributes
-@dataclass(frozen=True, kw_only=True)
+@dataclass(kw_only=True)
```

Comment:
Having immutable classes is useful. I don't remember by heart now how complex or simple this one is, but with dataclasses is always easy to create a new instance from another by [`replace()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.replace) function.

Not saying this is the right approach, just saying that there is a simple alternative to update immutable dataclasses.

Reply:
Dispatches are objects meant to be manipulated though, (except for the ones that execute it at the end of course). So most use cases become easier when you can modify it.

Reply:
Yeah, true.

Reply:
Well, actually not entirely.

I think the normal case is you create one, for which you don't need to mutate an existing one, and then you *consume* it, where you probably don't want to modify them either. For these 2 cases immutability seems like a nice safety feature.

Of course you could also update one, in that case using replace doesn't sound too cumbersome, and it is probably not a very common scenario, you probably made a mistake when creating it.

So IMHO having immutable `Dispatch` object might still be the best approach.

Reply:
Fair, I can live with immutable dispatch